src/irr/asset/CCPUMeshPacker.h

```diff
@@ -76,7 +76,10 @@ PackedMeshBufferData CCPUMeshPacker<MDIStructType>::alloc(const ICPUMeshBuffer c
 		}
 	}
 
-	auto MDIAllocAddr = m_MDIDataAlctr.alloc_addr(1u, 1u);
+	const uint32_t maxIdxCntPerPatch = m_maxTriangleCountPerMDIData * 3;
+	const uint32_t MDIStructsNeeded = std::ceil(static_cast<float>(meshBuffer->getIndexCount()) / maxIdxCntPerPatch);
```

Comment:
actually divide by the minimum index count per patch, so you have room for the extra splits (we can shrink the MDI allocation during `commit`, if we want)